cmd/kubeadm/app/componentconfigs/registrations.go

```diff
@@ -156,7 +150,9 @@ func (rs *Registrations) Default(internalcfg *kubeadmapi.ClusterConfiguration) {
 func (rs *Registrations) Validate(internalcfg *kubeadmapi.ClusterConfiguration) field.ErrorList {
 	allErrs := field.ErrorList{}
 	for kind, registration := range *rs {
-		allErrs = append(allErrs, registration.ValidateFunc(internalcfg, field.NewPath(string(kind)))...)
+		if registration.ValidateFunc != nil {
```

Comment:
Might be I'm missing something, but I don't see where ValidateFunc are defined...

Reply:
its in the registration for a component config?
https://github.com/kubernetes/kubernetes/blob/cdd186af984fd70315ecfca8dd0fd79c011a28ca/cmd/kubeadm/app/componentconfigs/registrations.go#L43

Reply:
Yes, the field is defined there, but I can't find where this value is initialised for kubelet/kube-proxy registrations

Reply:
They aren't ATM. I can set some funcs to do a pretty basic validation, but not the full blown, that the use of the internal types allowed us to do.
Hence, there are no validations ATM.

Reply:
strict unmarshal (warnings already enabled) + some basic validation should suffice.

Reply:
Currently, only strict unmarshalling is enabled. The problem with "basic validation" is that we don't know the definition of "basic" and at this point I don't want this to degrade to blatant copy-paste of the original component config validation funcs (which are usually quite large).

Reply:
i think this PR is OK and we can iterate.
/lgtm
/hold

@fabriziopandini @rosti 
feel free to hold cancel.